### PR TITLE
DDF-3470 Fix error with camel storage routes

### PR DESCRIPTION
**Pull request by troymohl**
#### What does this PR do?
Fixes errors where the Camel storage routes:
1. Updated PostIngestConsumer to start more threads and not constrain backlog.
2. Fixed camel creation errors when updating route definitions via metatype.
3. Updated Amazon S3 Secret Key field to be text. This avoids encoding empty strings as passwords for systems using roles on EC2 containers.

#### Who is reviewing it? 
@adimka  
@glenhein 

#### Select relevant component teams: 
https://github.com/orgs/codice/teams

#### Choose 2 committers to review/merge the PR. 
@bdeining
@millerw8

#### How should this be tested? (List steps with links to updated documentation)
Configure multiple Storage routes. 
Add/Remove/Update them
Ingest data and verify it is stored in the correct location(s).

#### What are the relevant tickets?
[DDF-3470](https://codice.atlassian.net/browse/DDF-3470)

#### Checklist:
- [ ] Documentation Updated
- [] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
For the general case, merging is gated on the following:

* Review by at least two contributors
* Review by at least two committers
* Successful CI
* Successful static analysis (SonarQube/Coverity/etc.)
* Heroing of change (details vary by ticket)

Certain exceptions may occur. For example, if there is a problem with the CI environment then a manually performed full build and static analysis check may be substituted _at the discretion of the repository maintainers_. Additionally, repository metadata such as this `PULL_REQUEST_TEMPLATE`, the repository `README`, and other non-functional items may be merged without requiring CI and static analysis.

Finally, an abbreviated review process may be in place for forward-port tickets immediately following branching and prior to branch divergence. In that case, approvals, CI, static analysis, and heroing that occur against the PR against the _release branch_ can be considered sufficient for reviewing the PR against the _master branch_.

Once the branches diverge, the process will return to normal.

Insofar as divergence is not a function of time but of change to the parent branch, the repository maintainers will determine when abbreviated review ends.

When abbreviated review is in effect, a note to that effect will be at the top of this template. Following is an example note:
```
##### ABBREVIATED REVIEW BETWEEN 2.12.X AND MASTER IS IN EFFECT
____

```

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/plugin/catalog-plugin-metacardbackup-filestorage/src/main/java/org/codice/ddf/catalog/plugin/metacard/backup/storage/filestorage/MetacardFileStorageRoute.java [hunk 1]
```diff
@@ -96,6 +109,11 @@ public void refresh(Map<String, Object> properties) throws Exception {
     super.refresh(properties);
   }
 
+  @Override
+  public List<String> getRouteIds() {
```
Comment (millerw8) [inline catalog/plugin/catalog-plugin-metacardbackup-filestorage/src/main/java/org/codice/ddf/catalog/plugin/metacard/backup/storage/filestorage/MetacardFileStorageRoute.java]: Does this result in these routes showing up when executing camel:route-list in the karaf console?

File: catalog/core/catalog-core-camelcomponent/src/main/java/ddf/camel/component/catalog/ingest/PostIngestConsumer.java [hunk 2]
```diff
@@ -48,28 +48,37 @@
...
-          StandardThreadFactoryBuilder.newThreadFactory("postIngestConsumerThread"));
+  private ExecutorService threadExecutor;
 
   private static final Logger LOGGER = LoggerFactory.getLogger(PostIngestConsumer.class);
 
   public PostIngestConsumer(CatalogEndpoint endpoint, Processor processor) {
     super(endpoint, processor);
     this.endpoint = endpoint;
+
+    Integer threadPoolSize =
```
Comment (glenhein) [inline catalog/core/catalog-core-camelcomponent/src/main/java/ddf/camel/component/catalog/ingest/PostIngestConsumer.java]: what if parseInt throws a NumberFormatException?

File: catalog/plugin/catalog-plugin-metacardbackup-filestorage/src/main/java/org/codice/ddf/catalog/plugin/metacard/backup/storage/filestorage/MetacardFileStorageRoute.java (hunk 1 above)
Comment (troymohl) [reply catalog/plugin/catalog-plugin-metacardbackup-filestorage/src/main/java/org/codice/ddf/catalog/plugin/metacard/backup/storage/filestorage/MetacardFileStorageRoute.java]: These routes do show up when executing `camel:route-list`

File: catalog/core/catalog-core-camelcomponent/src/main/java/ddf/camel/component/catalog/ingest/PostIngestConsumer.java (hunk 2 above)
Comment (troymohl) [reply catalog/core/catalog-core-camelcomponent/src/main/java/ddf/camel/component/catalog/ingest/PostIngestConsumer.java]: fixed.

**Comment by troymohl** [comment]
build now

File: catalog/plugin/catalog-plugin-metacardbackup-storage-common/src/main/java/org/codice/ddf/catalog/plugin/metacard/backup/common/MetacardStorageRoute.java [hunk 3]
```diff
@@ -77,9 +78,11 @@ public void stop(int code) {
       List<RouteDefinition> routesToRemove = new ArrayList<>();
       CamelContext context = getContext();
       for (RouteDefinition routeDefinition : context.getRouteDefinitions()) {
-        context.stopRoute(routeDefinition.getId());
-        routesToRemove.add(routeDefinition);
-        setRouteCollection(new RoutesDefinition());
+        if (getRouteIds().contains(routeDefinition.getId())) {
```
Comment (bdeining) [inline catalog/plugin/catalog-plugin-metacardbackup-storage-common/src/main/java/org/codice/ddf/catalog/plugin/metacard/backup/common/MetacardStorageRoute.java]: Will we benefit from adding a unit test for this condition?

Comment (troymohl) [reply catalog/plugin/catalog-plugin-metacardbackup-storage-common/src/main/java/org/codice/ddf/catalog/plugin/metacard/backup/common/MetacardStorageRoute.java]: I'm not sure a unit test would do more than artificially test if .contains() matches.

**Comment by mojogitoverhere** [comment]
:tada: Hero build was successful! :tada:

I did a quick build, started DDF, manually started and configured the `catalog-metacard-backup-filestorage` feature, and verified metacards were being backed up locally. I then updated the filestorage backup config to point to a different dir and verified metacards were still being backed up to the new dir.

I was also able to verify the S3 secret key text was not redacted like a password and was shown in plain text in the config UI. However, I was not able to verify the AWS access and secret keys were properly passed through due to this camel error: `AmazonS3Client must be specified`. A new ticket was written to address this issue https://codice.atlassian.net/browse/DDF-3491.  The metacards were backed up to S3 successfully using the instance role credentials (by leaving the AWS access and secret key fields blank in the config UI).

**Pull request merged**